Fixing Council pages,

davidwessman opened the pull request:
Removes all mentions of the_role actions, fixing old controllers.

Album-, workpost-controllers are not in used at the moment and I will not implement tests for them now.

davidwessman commented:
Fixing #148 and #149

davidwessman commented:
Did not have time to fix tests for this - would be great if someone can do some integration tests over the PageModule and PostModule.

davidwessman commented:
Should fix #145 and replace PR #146

davidwessman commented:
Fixes #148, #149, #151.

Still need some tests and Post-controller could need some work.
This should be tested on Dev - but it has some good changes which would be great to publish.

davidwessman commented:
@henrikssn 
We need to get this out soon.
I couldn't get deploy this to http://samson.fsektionen.se/projects/web/deploys/82, any clues?

henrikssn commented:
Fixat 974833b

henrikssn commented:
There seems to be another problem with the translations. I'll leave that one to you.

davidwessman commented:
@henrikssn 
Gonna try it out on dev, implemented test for #157  and fixed it.
Anything else?
(The function to give people a posts might not work correctly yet, but I started working on too much.)

davidwessman commented:
@henrikssn 
http://dev.fsektionen.se this error feels deploy-related, any clues?

davidwessman commented on config/routes.rb:
```diff
@@ -126,29 +125,30 @@
+          get '', action: :new, on: :collection, as: :new
+        end
+        resources :candidates, controller: 'elections/candidates',
+                  path: :kandidera, except: [:edit]
```
@fsek/admins 
How is this supposed to look?

jforberg replied:
I think the hound is wrong in this case. Turn off the flag if you can find it.

SirDirac commented:
This PR was deployed to Production. Reference: <a href='http://samson.fsektionen.se/projects/web/deploys/88' target='_blank'>83f5b59</a>

davidwessman commented:
The page link on the edit page doesn't work.
I think that is because I changed the page controller. Would need to fix this in the console.

jforberg commented on app/controllers/posts_controller.rb:
```diff
@@ -1,102 +1,74 @@
   def new
     @post = @council.posts.build
-    @councils = Council.order(title: :asc)
+    @post_permissions = @post.permissions.collect! { |p| p.id }
```
Yes. Also don't use the bang methods unless you have to. Even then, don't use the bang methods.

They modify their argument, after this line, @post.permissions will be a list of ints, not of permissions. Probably not what you want and in any case it's a code smell.

davidwessman replied:
@henrikssn Wrote this line, I'm not sure how it should be.

jforberg replied:
1. Use ``permissions = @post.permissions.map &:id

2. No need to do this in the controller, do it when you need the values instead (in the view). Also then you don't have to put it in an instance variable. In general try to set only a single instance variable in the controller.
